core/src/env/test_env.rs

```diff
@@ -399,6 +399,17 @@ impl TestEnvData {
     pub fn deposit_raw_event(&mut self, topics: &[Vec<u8>], data: &[u8]) {
         self.add_event(topics, data);
     }
+
+    #[allow(unused_variables)]
+    pub fn call<T: Decode>(
+        &mut self,
+        callee: &[u8],
+        gas: u64,
+        value: &[u8],
+        input_data: &[u8],
+    ) -> Option<T> {
+        unimplemented!();
```

Comment:
This should simply record the call by pushing a `struct CallData { callee, gas, value, input_data }` into a vector of those call data. See how it is done for events.